src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_2_PermanentWorkflow.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Common.DataModel;
+using Temporal.Worker.Activities;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    public class Part1_2_PermanentWorkflow
+    {
+        [Workflow(runMethod: nameof(SpeakUntilCancelledAsync))]
+        public class SayGreetingWorkflow
+        {            
+            private const int SingleWorkflowRunIterations = 10000;
+            private const string AddresseeNameDefault = "Boss";
+
+            private TaskCompletionSource<string> _setAddresseeName = new TaskCompletionSource<string>();
+            
+            public async Task SpeakUntilCancelledAsync(SpeechRequest greetingUtterance, WorkflowContext workflowCtx)
+            {
+                try
+                {
+                    for (int i = 0; i < SingleWorkflowRunIterations; i++)
+                    {
+                        string addresseeName = await GetAddresseeNameAsync();
+
+                        var greeting = new SpeechRequest($"{greetingUtterance.Text ?? "Hello"}, {addresseeName ?? AddresseeNameDefault}.");
+                        await workflowCtx.Orchestrator.Activities.ExecuteAsync(RemoteApiNames.Activities.SpeakGreeting, greeting);
+                    }
+                }
+                catch(TaskCanceledException tcEx)
+                {
+                    if (tcEx.Task == _setAddresseeName.Task)
+                    {
+                        return;
+                    }
+
+                    ExceptionDispatchInfo.Capture(tcEx).Throw();
+                }
+
+                workflowCtx.Orchestrator.ConfigureContinueAsNew(startNewRunAfterReturn: true, greetingUtterance);
+            }
+
+            private async Task<string> GetAddresseeNameAsync()
+            {
+                string addresseeName = await _setAddresseeName.Task;
+                _setAddresseeName = new TaskCompletionSource<string>();
+                return addresseeName;
+            }
+
+            [WorkflowSignalHandler]
+            public void SetAddressee(SpeechRequest input)
+            {
+                _setAddresseeName.TrySetResult(input?.Text);                
+            }
+
+            [WorkflowSignalHandler]
+            public void Cancel()
+            {
+                while (!_setAddresseeName.TrySetCanceled())
+                {
+                    _setAddresseeName = new TaskCompletionSource<string>();
+                }                
+            }
+        }
+
+        /// <summary>A sample activity implementation.</summary>
+        public static class Speak
+        {
+            public static Task GreetingAsync(SpeechRequest input, WorkflowActivityContext activityCtx)
+            {
```

Comment:
So this one looks very much like what I'd expect, without the manual serialization on either side, but I'm not clear on how this gets turned into an `IBasicActivity` implementer. Can you elaborate on that a bit?